module-bluetooth/Bluetooth/interface/profiles/HFP/HFP.cpp

```diff
@@ -453,99 +468,111 @@ namespace bluetooth
             codecsList.push_back(SCOCodec::mSBC);
             break;
         }
-        hfp_ag_init_codecs(codecsList.size(), reinterpret_cast<uint8_t *>(codecsList.data()));
+        hfp_ag_init_codecs(codecsList.size(), reinterpret_cast<std::uint8_t *>(codecsList.data()));
     }
-    void HFP::HFPImpl::initializeCall() const noexcept
-    {}
-    void HFP::HFPImpl::setAudioDevice(std::shared_ptr<bluetooth::BluetoothAudioDevice> audioDevice)
+
+    void HFP::HFPImpl::setAudioDevice(std::shared_ptr<bluetooth::BluetoothAudioDevice> audioDev)
     {
-        HFP::HFPImpl::audioDevice = std::static_pointer_cast<CVSDAudioDevice>(audioDevice);
+        HFP::HFPImpl::audioDevice = std::static_pointer_cast<CVSDAudioDevice>(audioDev);
         HFP::HFPImpl::audioDevice->setAclHandle(aclHandle);
         LOG_DEBUG("Audiodevice set!");
     }
-    void HFP::HFPImpl::startRinging() const noexcept
+
+    auto HFP::HFPImpl::incomingCallStarted() const noexcept -> Error::Code
     {
-        LOG_DEBUG("Starting incoming call");
-        currentCallStatus = CallStatus::Incoming;
+        LOG_DEBUG("Incoming call received");
         hfp_ag_incoming_call();
+        return Error::Success;
     }
-    void HFP::HFPImpl::stopRinging() const noexcept
+
+    auto HFP::HFPImpl::outgoingCallStarted(const std::string &number) const noexcept -> Error::Code
     {
-        LOG_DEBUG("Stop ringing called!");
+        LOG_DEBUG("Outgoing call placed");
+        hfp_ag_outgoing_call_initiated();
+        hfp_ag_outgoing_call_accepted();
+        hfp_ag_outgoing_call_ringing();
+        hfp_ag_send_phone_number_for_voice_tag(aclHandle, number.c_str());
+        return Error::Success;
     }
-    auto HFP::HFPImpl::callActive() const noexcept -> Error::Code
+
+    auto HFP::HFPImpl::incomingCallAnswered() const noexcept -> Error::Code
     {
-        if (currentCallStatus == CallStatus::Incoming) {
-            hfp_ag_answer_incoming_call(); // will answer the call if it wasn't answered
-        }
-        else {
-            hfp_ag_outgoing_call_established();
-        }
-        currentCallStatus = CallStatus::Active;
+        LOG_DEBUG("Incoming call answered");
+        hfp_ag_answer_incoming_call();
         return Error::Success;
     }
-    void HFP::HFPImpl::terminateCall() const noexcept
+
+    auto HFP::HFPImpl::outgoingCallAnswered() const noexcept -> Error::Code
     {
-        if (currentCallStatus == CallStatus::Active) {
-            hfp_ag_terminate_call();
-        }
-        else {
-            hfp_ag_call_dropped();
-        }
+        LOG_DEBUG("Outgoing call answered");
+        hfp_ag_outgoing_call_established();
+        return Error::Success;
```

Comment:
It is probably legacy code but it looks weird so I'd recommend using `Result::Success` instead of `Error` :)

Reply:
True, at first I also thought it looks weird, but then I just got used to it :P Fixed!